.github/workflows/ci.yml

```diff
@@ -32,6 +32,10 @@ on:
         required: true
         type: boolean
 
+concurrency:
+  group: pr-${{ github.ref }}
```

Comment:
This should be limited to PRs.

Reply:
It should also be scoped by workflow so we don't cancel a weekly build on main because a commit was pushed.

Reply:
Based on https://stackoverflow.com/questions/70928424/limit-github-action-workflow-concurrency-on-push-and-pull-request I would guess we should make it `group: ${{ github.head_ref || github.ref_name }}` and move this to `push_pr.yml`. 

I'll update the PR

Reply:
I've moved it out of this PR for now, created https://issues.apache.org/jira/browse/FLINK-30353 as follow up